selfservice/flow/registration/hook.go

```diff
@@ -217,7 +217,10 @@ func (e *HookExecutor) PostRegistrationHook(w http.ResponseWriter, r *http.Reque
 		Debug("Post registration execution hooks completed successfully.")
 
 	if a.Type == flow.TypeAPI || x.IsJSONRequest(r) {
-		e.d.Writer().Write(w, r, &APIFlowResponse{Identity: i})
+		e.d.Writer().Write(w, r, &APIFlowResponse{
+			Identity:     i,
+			ContinueWith: a.ContinueWith(),
+		})
```

Comment:
One more thing, I just thought about for this situation, is that all of this only applies to the API based interactions with these endpoints (SPA/Native).

We have not defined how this could work for browser based interactions. A few ideas:
1. Break backward compatibility and redirect to the verification flow here, if a flow was created as part of the registration flow.
2. Add a Query parameter to decide this behavior, that only applies to browser flows
3. Add a config option to set this.

Thoughts? cc @aeneasr

Reply:
Another thought, if the identity schema defines multiple verifiable addresses, we can't know where to redirect here. In that case, IMO, a `continue_with_address` query or even body param hint would be great.

Reply:
We decided to go with a query param added to the `return_to` redirect. This way we can send the verification flow ID to the UI, and it can then decide whether it wants to use it to do the verification, or continue and do the verification later. An existing implementation does not change.